stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -490,7 +492,13 @@ internal func += <Element, C : Collection>(
       from: lhs.firstElementAddress, count: oldCount)
     lhs.count = 0
     swap(&lhs, &newLHS)
-    (lhs.firstElementAddress + oldCount).initialize(from: rhs)
+    buf = UnsafeMutableBufferPointer(start: lhs.firstElementAddress + oldCount, count: numericCast(rhs.count))
+  }
+
+  if let remainder = buf.initialize(from: rhs) {
+    // there were elements that didn't fit in the existing buffer,
+    // append them in slow sequence-only mode
+    //FIXME: handle this possibility
```

Comment:
The PR title should have [WIP] if you don't actually intend to merge this as-is.

That said, why do you want to handle this case? Shouldn't we be consistent and make it a fatal error anywhere that Collection.count does not match the collections number of elements (assuming no intervening access to the Collection)?